src/App.tsx

```diff
@@ -1,27 +1,54 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleClick = useCallback(
+    async (goodsType: () => Promise<Good[]>) => {
+      try {
+        const goods = await goodsType();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+        setVisibleGoods(goods);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.log(error);
```

Comment:
Instead of console.log, consider displaying the error in the UI for better user experience 
<img width="1466" alt="Screenshot 2023-04-14 at 16 16 47" src="https://user-images.githubusercontent.com/53228259/232054224-7d8638d5-390e-4504-9ab4-84737a929f0c.png">
for example you can render something went wrong